pkg/cmd/factory/remote_resolver.go

```diff
@@ -49,7 +48,6 @@ func (rr *remoteResolver) Resolver(hostOverride string) func() (context.Remotes,
 		}
 
 		knownHosts := map[string]bool{}
-		knownHosts[ghinstance.Default()] = true
```

Comment:
Removing this line leads to a regression in this case:
1. When in a repository cloned from `github.com` and
2. Only GHE is authenticated in `hosts.yml`

Running a command such as `gh issue list` will print an error:
> none of the git remotes configured for this repository point to a known GitHub host. To tell gh about a new GitHub host, please use `gh auth login`

It might be confusing to say to a user that `github.com` is not a known GitHub host; hence we had this line that special-cases `github.com`.

I think the change in logic should only affect the case when there is multiple git remotes for different hosts, but we want to default to the GHE hostname that is already authenticated to. Otherwise, we can keep recognizing `github.com` as a known GitHub host.